homeassistant/components/shelly/entity.py

```diff
@@ -444,7 +442,7 @@ def available(self) -> bool:
         """Available."""
         available = super().available
 
-        if not available or not self.entity_description.available:
+        if not available or not self.entity_description.available or self.block is None:
```

Comment:
We allow setting conditions to mark the device unavailable based on data from device blocks - https://github.com/home-assistant/core/blob/2d636ad4d95daac547edc0e1ea134f3e0d8c8fbd/homeassistant/components/shelly/number.py#L46
This bug existed before this change but was not covered in tests since we only tested few sensors which did not use this logic and number platform was not using the entity description when the device is restored. After fixing entity descriptions for number platform the test discovered this bug.